rotkehlchen/chain/evm/transactions.py

```diff
@@ -604,12 +604,14 @@ def ensure_tx_data_exists(
         - RemoteError if there is a problem querying the data sources or transaction hash does
         not exist.
         """
-        query, bindings = EvmTransactionsFilterQuery.make(tx_hash=tx_hash, chain_id=self.evm_inquirer.chain_id).prepare()  # noqa: E501
-        query, bindings = self.dbevmtx._form_evm_transaction_dbquery(query=query, bindings=bindings, has_premium=True)  # noqa: E501
-        tx_data = cursor.execute(query, bindings).fetchone()
+        transactions = self.dbevmtx.get_evm_transactions(
+            cursor=cursor,
+            filter_=EvmTransactionsFilterQuery.make(tx_hash=tx_hash, chain_id=self.evm_inquirer.chain_id),  # noqa: E501
+            has_premium=True,
+        )
```

Comment:
shouldn't you query inside the if statement once you know that the transaction is in the database? You don't seem to use this query result more